vscode/client/src/server.ts

```diff
@@ -0,0 +1,238 @@
+					if (!(await installBinaryViaCargoInstall())) {
+						await window.showErrorMessage(
+							"Failed to update Pest Language Server."
+						);
+					}
+				}
+			}
+		} catch (_) {
+			outputChannel.appendLine("[TS] Failed to run update check.");
+		}
+	}
+
+	return path;
+}
+
+async function findServerPath(): Promise<string | undefined> {
+	const config = workspace.getConfiguration("pestIdeTools");
+
+	// Check for custom server path
+	if (config.get("serverPath")) {
+		return config.get("serverPath") as string;
+	}
+
+	const cargoBinDirectory = getCargoBinDirectory();
+
+	if (!cargoBinDirectory) {
+		outputChannel.appendLine("[TS] Could not find cargo bin directory.");
+		return undefined;
+	}
+
+	const expectedPath = join(cargoBinDirectory, getExpectedBinaryName());
+	outputChannel.appendLine(`[TS] Trying path ${expectedPath}...`);
+
+	if (await checkValidity(expectedPath)) {
+		return expectedPath;
+	}
+
+	const choice = await window.showWarningMessage(
+		"Failed to find an installed Pest Language Server. Would you like to install one using `cargo install`?",
+		{},
+		"Yes"
+	);
+
+	if (!choice) {
+		outputChannel.appendLine("[TS] Not installing server.");
+		return undefined;
+	}
+
+	if (await installBinaryViaCargoInstall()) {
+		return expectedPath;
+	} else {
+		await window.showErrorMessage(
+			"Failed to install Pest Language Server. Please either run `cargo install pest-language-server`, or set a custom path using the configuration `pestIdeTools.serverPath`."
+		);
+	}
+
+	return undefined;
+}
+
+function getCargoBinDirectory(): string | undefined {
+	const cargoInstallRoot = process.env["CARGO_INSTALL_ROOT"];
+
+	if (cargoInstallRoot) {
+		return cargoInstallRoot;
+	}
+
+	const cargoHome = process.env["CARGO_HOME"];
+
+	if (cargoHome) {
+		return join(cargoHome, "bin");
+	}
+
+	let home = process.env["HOME"];
+
+	if (process.platform === "win32") {
+		home = process.env["USERPROFILE"];
+	}
+
+	if (home) {
+		return join(home, ".cargo", "bin");
+	}
+
+	return undefined;
+}
+
+function getExpectedBinaryName(): string {
+	switch (process.platform) {
+		case "win32":
+			return "pest-language-server.exe";
+		default:
+			return "pest-language-server";
+	}
+}
+
+async function checkValidity(path: string): Promise<boolean> {
+	try {
+		await stat(path);
+		return true;
+	} catch (_) {
+		return false;
+	}
+}
+
+async function installBinaryViaCargoInstall(): Promise<boolean> {
+	outputChannel.appendLine("[TS] Installing server.");
+
+	return await window.withProgress(
+		{
+			location: ProgressLocation.Notification,
+			cancellable: false,
+			title: "Installing Pest Language Server",
+		},
+		async (progress, _) => {
+			try {
+				progress.report({ message: "Spawning `cargo install` command" });
+
+				const process = spawn(
+					"cargo",
+					[
+						"install",
+						"--git",
+						"https://github.com/pest-parser/pest-ide-tools",
+						"--branch",
+						"feat/tower-lsp",
+						"--bin",
+						"pest-language-server",
```

Comment:
Don't forget release `pest_language_server` crate first, and change there to `cargo install pest_language_server`

Reply:
Yes, we will get this done soon